src/Sylius/Bundle/TaxonomyBundle/Doctrine/ORM/TaxonRepository.php

```diff
@@ -166,4 +166,56 @@ protected function createTranslationBasedQueryBuilder(?string $locale): QueryBui
 
         return $queryBuilder;
     }
+
+    public function findOneAbove(TaxonInterface $taxon): ?TaxonInterface
+    {
+        $query = $this->createQueryBuilder('o');
+        $taxonParent = $taxon->getParent();
+
+        if (null !== $taxonParent) {
+            $query
+                ->andWhere('o.parent = :parent')
+                ->setParameter('parent', $taxonParent)
+            ;
+        } else {
+            $query
+                ->andWhere('o.parent IS NULL')
+            ;
+        }
+
+        return $query
+            ->andWhere('o.position < :taxonPosition')
+            ->setParameter('taxonPosition', $taxon->getPosition())
+            ->addOrderBy('o.position', 'DESC')
+            ->setMaxResults(1)
```

Comment:
Is it necessary in presence of `getOneOrNullResult`? Which will throw exception like so

```php
        if (count($result) > 1) {
            throw new NonUniqueResultException();
        }
```

Reply:
I wanted to eliminate a need to handle the exception.